### PR TITLE
zscaler_zia: Add eurl hex-encoded field for url parsing in web logs.

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally
`$ cd packages/zscaler_zia && eval "$(elastic-package stack shellinit)" && elastic-package test pipeline --generate -v --data-streams=web`
```
--- Test results for package: zscaler_zia - START ---
╭─────────────┬─────────────┬───────────┬───────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE     │ DATA STREAM │ TEST TYPE │ TEST NAME                                             │ RESULT │ TIME ELAPSED │
├─────────────┼─────────────┼───────────┼───────────────────────────────────────────────────────┼────────┼──────────────┤
│ zscaler_zia │ web         │ pipeline  │ (ingest pipeline warnings test-web-http-endpoint.log) │ PASS   │ 448.204792ms │
│ zscaler_zia │ web         │ pipeline  │ (ingest pipeline warnings test-web.log)               │ PASS   │    607.004ms │
│ zscaler_zia │ web         │ pipeline  │ test-web-http-endpoint.log                            │ PASS   │ 132.522542ms │
│ zscaler_zia │ web         │ pipeline  │ test-web.log                                          │ PASS   │ 172.145542ms │
╰─────────────┴─────────────┴───────────┴───────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: zscaler_zia - END   ---
Done
```

### CODE REVIEW

File: packages/zscaler_zia/_dev/build/docs/README.md [hunk 1]
```diff
@@ -216,12 +216,12 @@ See: [Zscaler Vendor documentation](https://help.zscaler.com/zia/nss-feed-output
...
...size":"%d{reqdatasize}","respheadersize":"%d{resphdrsize}","respdatasize":"%d{respdatasize}","riskscore":"%d{riskscore}","ruletype":"%s{ruletype}","second":"%02d{ss}","srvcertchainvalpass":"%s{srvcertchainvalpass}","srvcertvalidationtype":"%s{srvcertvalidationtype}","srvcertvalidityperiod":"%s{srvcertvalidityperiod}","srvsslcipher":"%s{srvsslcipher}","serversslsessreuse":"%s{serversslsessreuse}","srvocspresult":"%s{srvocspresult}","srvtlsversion":"%s{srvtlsversion}","srvwildcardcert":"%s{srvwildcardcert}","ssldecrypted":"%s{ssldecrypted}","throttlereqsize":"%d{throttlereqsize}","throttlerespsize":"%d{throttlerespsize}","totalsize":"%d{totalsize}","trafficredirectmethod":"%s{trafficredirectmethod}","unscannabletype":"%s{unscannabletype}","upload_doctypename":"%s{upload_doctypename}","upload_fileclass":"%s{upload_fileclass}","upload_filetype":"%s{upload_filetype}","urlcatmethod":"%s{urlcatmethod}","urlsubcat":"%s{urlcat}","urlsupercat":"%s{urlsupercat}","urlclass":"%s{urlclass}","useragentclass":"%s{uaclass}","useragenttoken":"%s{ua_token}","userlocationname":"%s{userlocationname}","year":"%04d{yyyy}","ztunnelversion":"%s{ztunnelversion}","zpa_app_seg_name":"%s{zpa_app_seg_name}"\}\}
```
Comment (efd6) [inline packages/zscaler_zia/_dev/build/docs/README.md:219]: Diff is `lsversion}",+{"eurl":"%s{eurl}",}"url":"%s{ur`.

File: packages/zscaler_zia/_dev/build/docs/README.md [hunk 2]
```diff
@@ -216,12 +216,12 @@ See: [Zscaler Vendor documentation](https://help.zscaler.com/zia/nss-feed-output
...
...lelabel":"4951704103","ozpa_app_seg_name":"7648246731","externalsslpolicyreason":"Blocked","productversion":"5.0.902.95524_04","rdr_rulename":"FWD_Rule_1","refererhost":"www.example.com for http://www.example.com/index.html","reqheadersize":300,"reqdatasize":1000,"respheadersize":500,"respdatasize":10000,"riskscore":10,"ruletype":"File Type Control","second":48,"srvcertchainvalpass":"Unknown","srvcertvalidationtype":"EV (Extended Validation)","srvcertvalidityperiod":"Short","srvsslcipher":"SSL3_CK_RSA_NULL_MD5","serversslsessreuse":"Unknown","srvocspresult":"Good","srvtlsversion":"SSL2","srvwildcardcert":"Unknown","ssldecrypted":"Yes","throttlereqsize":5,"throttlerespsize":7,"totalsize":11800,"trafficredirectmethod":"DNAT (Destination Translation)","unscannabletype":"Encrypted File","upload_doctypename":"Corporate Finance","upload_fileclass":"upload_fileclass","upload_filetype":"RAR Files","urlcatmethod":"Database A","urlsubcat":"Entertainment","urlsupercat":"Travel","urlclass":"Bandwidth Loss","useragentclass":"Firefox","useragenttoken":"Google Chrome (0.x)","userlocationname":"userlocationname","year":2023,"ztunnelversion":"ZTUNNEL_1_0","zpa_app_seg_name":"ZPA_test_app_segment"}}
```
Comment (efd6) [inline packages/zscaler_zia/_dev/build/docs/README.md:224]: Diff is `on":"SSL2",+{"eurl":"www.trythisencodeurl.com/index",}"url":"ww`

**Review by efd6 — COMMENTED** [review]
I'd like to see a test case where there is a URL that has been conventionally encoded, preferably from a real event if possible.

File: packages/zscaler_zia/data_stream/web/_dev/test/pipeline/test-web.log [hunk 3]
```diff
@@ -1,5 +1,4 @@
...
+
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/zscaler_zia/data_stream/web/_dev/test/pipeline/test-web.log]: Removed

File: packages/zscaler_zia/_dev/build/docs/README.md (hunk 2 above)
Comment (kcreddy) [reply packages/zscaler_zia/_dev/build/docs/README.md:224]: Not sure how to check this. I did a git diff and returned without the extra `{` `}`

File: packages/zscaler_zia/_dev/build/docs/README.md (hunk 1 above)
Comment (kcreddy) [reply packages/zscaler_zia/_dev/build/docs/README.md:219]: Same here.
https://github.com/elastic/integrations/pull/10657#discussion_r1698289542

**Comment by kcreddy** [comment]
> I'd like to see a test case where there is a URL that has been conventionally encoded, preferably from a real event if possible.

@efd6 I added a new log line which simulates an anonymised `url` similar to a real event.

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**